brmp/pyro_backend.py

```diff
@@ -119,24 +127,36 @@ def nuts(data, model, iter=None, warmup=None, num_chains=None):
     nuts_kernel = NUTS(model.fn, jit_compile=False, adapt_step_size=True)
     mcmc = MCMC(nuts_kernel, num_samples=iter, warmup_steps=warmup, num_chains=num_chains)
     mcmc.run(**data)
-    samples = mcmc.get_samples()
+
+    samples = mcmc.get_samples(group_by_chain=True)
+    # Pyro doesn't insert a chain dim when num_chains==1.
```

Comment:
We should do this in Pyro too. At one point, we didn't have the `group_by_chain` keyword argument in Pyro, and it made sense to not insert a dim for the default `num_chains=1`, but now that we have this argument which is set to False by default, we should not make any distinctions for `num_chains=1`.

Reply:
I will fix this shortly in Pyro. If you notice any such odd differences between the two, let us know!